File: TaggyAppBackend/TaggyAppBackend.Api/Models/Dtos/File/GetFileDto.cs
using TaggyAppBackend.Api.Models.Dtos.Tag;

namespace TaggyAppBackend.Api.Models.Dtos.File;

public class GetFileDto : GetDto
{
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public string Url { get; set; } = null!;
    public string ContentType { get; set; } = null!;
    public long Size { get; set; }

    public string CreatorId { get; set; } = null!;
    public string GroupId { get; set; } = null!;
    public List<GetTagDto> Tags { get; set; } = [];
}

File: TaggyAppFrontend/src/app/models/dtos/file/getFileDto.ts
import { GetDto } from "../getDto";
import { GetTagDto } from "../tag/getTagDto";

export class GetFileDto extends GetDto {
  name!: string;
  description?: string;
  url!: string;
  contentType!: string;
  size!: number;

  creatorId!: string;
  groupId!: string;
  tags!: GetTagDto[];
}
